pkg/controller/tsp/tsp_controller.go

```diff
@@ -139,17 +105,18 @@ func (tc *Controller) syncTimeSeriesPrediction(prediction *predictionv1alph1.Tim
 	}
 
 	tc.tsPredictionMap.Store(key, prediction)
-	// add the prediction to time delay queue for update
-	tc.delayQueue.AddAfter(key, time.Duration(prediction.Spec.PredictionWindowSeconds)*time.Second)
+
+	return tc.syncPredictionStatus(ctx, prediction)
 
 }
 
 func NewMetricContext(prediction *predictionv1alph1.TimeSeriesPrediction) *predconfig.MetricContext {
 	c := &predconfig.MetricContext{
 		Namespace:  prediction.Namespace,
 		TargetKind: prediction.Spec.TargetRef.Kind,
+		Name:       prediction.Spec.TargetRef.Name,
 	}
-	if c.TargetKind != predconfig.TargetKindNode && prediction.Spec.TargetRef.Namespace != "" {
+	if strings.ToLower(c.TargetKind) != predconfig.TargetKindNode && prediction.Spec.TargetRef.Namespace != "" {
```

Comment:
I see the actual const is Node, a toLower will make this as false all the time?
const TargetKindNode = "Node"